logdna/resource_view.go

```diff
@@ -99,49 +101,69 @@ func buildChannels(emailchannels []interface{}, pagerdutychannels []interface{},
 		url := i["url"].(string)
 
 		headersMap := make(map[string]string)
-		templateMap := make(map[string]string)
 
 		for k, v := range headers {
 			headersMap[k] = v.(string)
 		}
 
-		for k, v := range bodytemplate {
-			templateMap[k] = v.(string)
+		var webhook channel
+		if bodytemplate != "" {
+			var bt map[string]interface{}
+			err := json.Unmarshal([]byte(bodytemplate), &bt)
+
+			if err != nil {
+				return channels, err
+			}
```

Comment:
fair amount of duplication below in the latest revision. you can create the `webhook` with all the common properties, then if there's a body template, add it to the struct:

```golang
webhook := channel{
   Headers: headersMap
   Immediate: immediate
   ...
}

if bodytemplate != "" {
  var bt map[string]interface{}
  err := json.Unmarshal([]byte(bodytemplate), &bt)

  if err != nil {
    return channels, err
  }
  webhook.BodyTemplate = bt
}